File: SampleProjects/GoT-Characters-master/Objective-C/GoT/UIView+Additions.m
//
//  UIView+Additions.m
//  ParallaxDemo
//
//  Created by Paciej on 18/10/15.
//  Copyright (c) 2015 Paciej. All rights reserved.
//

#import "UIView+Additions.h"

@implementation UIView (Additions)

+ (void)addEdgeConstraint:(NSLayoutAttribute)edge
                superview:(UIView *)superview
                  subview:(UIView *)subview {
    [superview
     addConstraint:[NSLayoutConstraint constraintWithItem:subview
                                                attribute:edge
                                                relatedBy:NSLayoutRelationEqual
                                                   toItem:superview
                                                attribute:edge
                                               multiplier:1
                                                 constant:0]];
}

+ (void)addConstraintsFromStrings:(NSArray *)strings
                      withMetrics:(NSDictionary *)metrics
                         andViews:(NSDictionary *)views
                           toView:(UIView *)view {
    NSMutableArray *constraints = [NSMutableArray new];
    for (NSString *string in strings) {
        [constraints addObjectsFromArray:[NSLayoutConstraint
                                          constraintsWithVisualFormat:string
                                          options:0
                                          metrics:metrics
                                          views:views]];
    }
    [view addConstraints:constraints];
}

+ (instancetype)autolayoutView {
    UIView *view = [[[self class] alloc] initWithFrame:CGRectZero];
    view.translatesAutoresizingMaskIntoConstraints = NO;
    return view;
}

@end

File: SampleProjects/GoT-Characters-master/Swift/GoT/UIView+Additions.swift
//
//  UIView+Additions.swift
//  ParallaxDemo
//
//  Created by Paciej on 18/10/15.
//  Copyright (c) 2015 Paciej. All rights reserved.
//

import UIKit

extension UIView {
    /** Creates constraints from array of visual format language strings and adds
     * them to the view*/
    class func addConstraints(fromStrings strings: [Any]?, withMetrics metrics: [AnyHashable : Any]?, andViews views: [AnyHashable : Any]?, to view: UIView?) {
        var constraints = [AnyHashable]()
        for string: String? in strings as? [String?] ?? [] {
            if let aViews = views as? [String : Any] {
                constraints.append(contentsOf: NSLayoutConstraint.constraints(withVisualFormat: string ?? "", options: [], metrics: metrics as? [String : Any], views: aViews))
            }
        }
        if let aConstraints = constraints as? [NSLayoutConstraint] {
            view?.addConstraints(aConstraints)
        }
    }

    //* Creates view ready-to-use with auto layout
    class func autolayout() -> Self {
        let view = self.init(frame: CGRect.zero)
        view.translatesAutoresizingMaskIntoConstraints = false
        return view
    }

    class func addEdgeConstraint(_ edge: NSLayoutConstraint.Attribute, superview: UIView?, subview: UIView?) {
        if let aSubview = subview {
            superview?.addConstraint(NSLayoutConstraint(item: aSubview, attribute: edge, relatedBy: .equal, toItem: superview, attribute: edge, multiplier: 1, constant: 0))
        }
    }
}